Feat/#10

hyundang opened the pull request:
챌린지 api 구현 완료

hyundang committed:
chore: ejs 설정

hyundang committed:
feat: challenge 등록 구현

hyundang committed:
refactor: model import 수정

hyundang committed:
feat: controller dto 추가

hyundang committed:
fix: User Model에 img 필드 추가

hyundang committed:
fix: Comment, Post model 변경

- comments 필드 추가

hyundang committed:
feat: 댓글 등록 구현

hyundang committed:
fix: Error [ERR_HTTP_HEADERS_SENT] 해결

hyundang committed:
fix: Model 변경

- Badge, User, UserInfo

hyundang committed:
feat: 좋아요 등록 구현

hyundang committed:
feat: 스크랩 등록 구현

hyundang committed:
style: 주석 변경

hyundang committed:
style: 변수명 변경

hyundang committed:
fix: model 수정

- Challenge, Concert에서  timestamp 삭제
- Challenge에서 generation 삭제
- Post에 generation 추가

hyundang committed:
feat: commentDTO 추가

hyundang committed:
fix: model 변경사항 반영, commentDTO 추가

hyundang committed:
feat: 챌린지 전체 불러오기 구현

hyundang committed:
fix: model에서 timestamp 삭제

hyundang committed:
fix: Badge timestamp 추가

hyundang committed:
Merge remote-tracking branch 'origin/develop' into feat/#10

hyundang committed:
feat: userinfo 삭제

hyundang committed:
Merge branch 'develop' into feat/#10

hyundang committed:
style: 주석, 줄바꿈

hyundang committed:
feat: challenge api 구현

- 챌린지 검색 get
- 챌린지 상세 정보 get
- 챌린지 수정 patch

hyundang committed:
fix: join -> require 추가

hyundang committed:
fix: challenge patch api 수정

- 데이터 업데이트 부분 수정

hyundang committed:
feat: challenge api 구현

- 회고 삭제
- 좋아요 취소
- 스크랩 취소

hyundang committed:
fix: 전체 조회

- required 추가

holmir97 commented on src/service/challengeService.ts:
```diff
@@ -311,82 +311,123 @@ export const postScrap = async (challengeID: number, userID: number) => {
+      isDeleted: false,
+      generation,
+    },
+    include: [
```
아직 inner join 반영 안한거죠?

holmir97 replied:
앗..! 다음 커밋에서 변경사항 확인했어요!

holmir97 commented on src/service/challengeService.ts:
```diff
@@ -311,82 +311,123 @@ export const postScrap = async (challengeID: number, userID: number) => {
+  });
+
+  const resData: challengeDTO.getChallengeAllResDTO[] = await Promise.all(
+    challengeList.map(async (challenge) => {
```
시퀄라이즈 한 객체는 object형식으로 반환이 되던데 map함수의 작동이 정상적인가요?

hyundang replied:
response 제대로 오는거 테스트해보긴 했는데 다시 한번 테스트 해서 확인해볼게요!!

holmir97 replied:
넵! 혹시 안된다면 Object.values(바꿀 object)로 한번에 변경 및 사용 가능합니다!

hyundang replied:
제대로 오는것 같습니다!!

holmir97 commented on src/service/challengeService.ts:
```diff
@@ -428,228 +420,285 @@ export const getChallengeAll = async (
+ *  @error
+ *    1. challenge id가 없을 때
+ */
+
```
회고를 하나만 가져오는 경우에는 isLike isSrap이 필요없나요?

hyundang replied:
앗 빼먹었네요 감삼당!!

xxeol2 commented:
확인완료⚡️

hyundang committed:
fix: Get challenge/:challengeID 

- isLike, isScrap 추가

holmir97 commented on src/service/challengeService.ts:
```diff
@@ -105,6 +106,214 @@ export const postChallenge = async (
+    await badge.save();
+  }
+
+  const resData: challengeDTO.postCommentResDTO = {
```
기존에는 결과까지 응답 메시지로 보냈었는데! 리팩토링 과정이니 post에서는 결과는 빼는것이 어떨까요!

holmir97 commented on src/service/challengeService.ts:
```diff
@@ -105,6 +106,214 @@ export const postChallenge = async (
+  };
+
+  return resData;
+};
```
기존에는 결과까지 응답 메시지로 보냈었는데! 리팩토링 과정이니 post에서는 결과는 빼는것이 어떨까요!

hyundang replied:
좋은 것 같슴당!!

holmir97 commented on src/service/challengeService.ts:
```diff
@@ -105,6 +106,214 @@ export const postChallenge = async (
+      order: parentComment.groupNum,
+    });
+    await Comment.increment("groupNum", { by: 1, where: { id: parentID } });
+
```
groupNum 이 댓글과 자식 댓글의 한 그룹을 나타내는 거라면 댓글의 groupNum이 올라가도록 해야할 것 같습니다. 제가 이해한 것이 맞나요?

hyundang replied:
groupNum은 한 그룹 안에 들어있는 대댓글 전체 개수 입니당~~
근데 생각해보니까 이 부분은 없어도 될 것 같다는 생각이 드네여 왜 넣었을까요..??

hyundang replied:
앗 이거 댓글 순서 때문에 있어야 합니당

hyundang committed:
fix: post comment -> res data 삭제

hyundang committed:
fix: 8/23일자 수정사항 반영

- Post model에 userLikes, userScraps 추가
- Comment model에 children, comment 추가
- Hashtag model 삭제
- 댓글 파싱 로직 변경
- 댓글 join 수정(required: false, order 추기)
- isLikes, isScraps 로직 수정

hyundang commented:
@holmir97 @xxeol2 서버 열어분 코드 달라진 부분 체크하고 리뷰 해주세영🤗

**변경사항**
--------------
- Post, Comment 모델 수정
- 댓글 조인 부분 수정(order, required 추가)
- 댓글 파싱 로직 수정
- userID에 따른 isScrap, isLike 포함 여부
